scripts/check-events

```diff
@@ -155,30 +156,67 @@ class Event
       end
     end
 
-    has_errors = handle.logs.one? { |type, _, _| type == 'error' }
+    evt_name_events_dropped = 'component_discarded_events_total'
+    evt_name_error = 'component_errors_total'
 
-    # Make sure Error events output an error
-    if has_errors or @name.end_with? 'Error'
+    has_error_logs = handle.logs.one? { |type, _, _| type == 'error' }
+
+    is_events_dropped_evt = (@name.end_with? 'EventsDropped' or @counters.include? evt_name_events_dropped)
+
+    # Validate <Name>Error events
+    if (has_error_logs and !is_events_dropped_evt) or @name.end_with? 'Error'
+
+      # Name check
       append('Error events MUST be named "___Error".') unless @name.end_with? 'Error'
+      # Outputs an error log
       handle.logs_should_have('error')
-      counters_must_include('component_errors_total', ['error_type', 'stage'])
-    end
+      # Metric check
+      counters_must_include(evt_name_error, ['error_type', 'stage'])
+
+      # Make sure Error events contain the required parameters
+      handle.logs.each do |type, message, parameters|
+        if type == 'error'
+          ['error_type', 'stage'].each do |parameter|
+            unless parameters.include? parameter
+              @reports.append("Error log for Error event MUST include parameter \"#{parameter}\".")
+            end
+          end
 
-    # Make sure error events contain the right parameters
-    handle.logs.each do |type, message, parameters|
-      if type == 'error'
-        ['error_type', 'stage'].each do |parameter|
-          unless parameters.include? parameter
-            @reports.append("Error log MUST include parameter \"#{parameter}\".")
+          ['error_code', 'error_type', 'stage'].each do |parameter|
+            if parameters.include? parameter and !@counters[evt_name_error].include? parameter
+              @reports.append("Counter \"#{evt_name_error}\" must include \"#{parameter}\" to match error log.")
+            end
           end
         end
+      end
+    end
+
+    # TODO remove @skip_dropped_events check logic after DroppedEvents audit is complete
+    # (https://github.com/vectordotdev/vector/issues/13995)
+
+    # Validate <Namespace>EventsDropped events
+    if is_events_dropped_evt && !@skip_dropped_events
+
+      # Name check
+      append('EventsDropped events MUST be named "___EventsDropped".') unless @name.end_with? 'EventsDropped'
```

Comment:
```suggestion
      append('Events-dropped events MUST be named "___EventsDropped".') unless @name.end_with? 'EventsDropped'
```

Reply:
Was trying to closely follow https://github.com/vectordotdev/vector/blob/master/docs/specs/instrumentation.md#eventsdropped 

, what about this ? `<Namespace>EventsDropped`

Reply:
Oh, my suggestion wasn't about the name of the event, but about the text describing it, to parallel the error event message:
```
append('Error events MUST be named "___Error".') unless @name.end_with? 'Error'
```

Reply:
Ah, right sorry I wasn't clear- I meant that for the text describing it.
I figured if the spec defines the "class" of events as `Error` events and `EventsDropped` events, then the text displayed if this check fails should follow that ?

ie, keep the original I have , or change the line to:

```Ruby
append('<Namespace>EventsDropped events MUST be named "___EventsDropped".') unless @name.end_with? 'EventsDropped'
```
, Though I think my original is less confusing.

Reply:
I'm fine either way, just wanted to make sure you understood the reason I chose "EventsDropped" was to try to align with the spec wording.

Reply:
Yeah, that's fine, I won't block on that.